src/ui/widgets/Checkbox/checkbox.tsx

```diff
@@ -31,33 +37,37 @@ export type CheckboxComponentProps = InferWidgetProps<typeof CheckboxProps> &
 export const CheckboxComponent = (
   props: CheckboxComponentProps
 ): JSX.Element => {
-  const style = {
-    ...commonCss(props as any),
-    display: "flex",
-    alignItems: "center",
-    cursor: "pointer"
-  };
-
   const [checked, setChecked] = useState(true);
 
-  const toggle = (): void => {
+  const handleChange = (): void => {
     setChecked(!checked);
   };
-  const inp = (
-    <input
-      style={{ cursor: "inherit" }}
-      id="cb"
-      type="checkbox"
-      checked={checked}
-      readOnly={true}
-    />
-  );
 
   return (
-    <form onClick={toggle} style={style}>
-      {inp}
-      <label style={{ cursor: "inherit" }}>{props.label}</label>
-    </form>
+    <ThemeProvider theme={defaultColours}>
+      <FormControlLabel
+        sx={{
+          color:
+            props.foregroundColor?.toString() ??
+            defaultColours.palette.primary.contrastText,
+          "&:npt($checked) .MuiIconButton-label:after": {
+            color:
+              props.foregroundColor?.toString() ??
+              defaultColours.palette.primary.main
+          }
+        }}
```

Comment:
A few notes here:

- The theme colours don't match Phoebus defaults, as mentioned in the other PR as well
- For the label, it would be good to have word wrapping and font settings (This is something that was not included in the old version but should have been, so would be good to add in)